pkg/exporter/sumologicexporter/sender.go

```diff
@@ -241,6 +247,34 @@ func (s *sender) sendLogs(ctx context.Context, flds fields) ([]pdata.LogRecord,
 	return droppedRecords, nil
 }
 
+// sendLogs sends log records from the logBuffer in OTLP format
+// and as the result of execution
+// returns array of records which has not been sent correctly and error
+// TODO: add support for HTTP limits
+func (s *sender) sendOTLPLogs(ctx context.Context, flds fields) ([]pdata.LogRecord, error) {
+	ld := pdata.NewLogs()
+	rls := ld.ResourceLogs()
+	rls.Resize(1)
+	rl := rls.At(0)
+	ills := rl.InstrumentationLibraryLogs()
+	ills.Resize(1)
+	ill := ills.At(0)
+	logs := ill.Logs()
```

Comment:
preallocate?

Reply:
Added. Also I used `CopyTo` method as its recommended to not reuse LogRecord in multiple structures